### ISSUE TITLE
Discrepancy results between online demo and SamAutomaticMaskGenerator

### ISSUE THREAD

**Issue by michelbotros** [facebookresearch/segment-anything#716]
Hi,

I have been trying out SAM in the online demo and it works great:

**Image:**
![image](https://github.com/facebookresearch/segment-anything/assets/4972463/81a57237-08fb-4675-82b5-3e69a0b28eef)

**Result demo:**
![results_demo](https://github.com/facebookresearch/segment-anything/assets/4972463/34f4b35f-55fd-400c-8648-bd87587cfd98)

I then tried to recreate these results with the code using the SamAutomaticMaskGenerator and a vit-h (default) model. However, I get a different result. Large portion of the background is segmented and there is also many more small regions detected.

**Result code:**
![results_code](https://github.com/facebookresearch/segment-anything/assets/4972463/29f32f2f-0b93-4622-ba2b-187bce016daf)

For my application the demo result is preferred. If anyone knows how to recreate the demo results, it would be greatly appreciated if its shared!

**Comment by MichaelP84** [comment]
I am experiencing the same thing, just between the online demo and the workflow showed in predictor_example.ipynb on my own images

**Comment by Nivratti** [comment]
The primary reason for the variation in masks between the online demo and the local notebook is that the online demo uses a different model for "segment everything" functionality.  The online demo is powered by the model hosted at the following endpoint:

[segment_everything_box_model](https://model-zoo.metademolab.com/predictions/segment_everything_box_model)

**Comment by CurryKd7** [comment]
> 在线演示和本地笔记本之间掩码差异的主要原因是，在线演示使用不同的模型来实现“分割所有内容”功能。在线演示由托管在以下终端节点的模型提供支持：
> 
> [segment_everything_box_model](https://model-zoo.metademolab.com/predictions/segment_everything_box_model)

The link you provided cannot be opened, why

**Comment by pzhren** [comment]
Indeed, the effect of online demo is obviously better than local, why?

**Comment by scchess** [comment]
There is no such thing as free lunch, the best tuned model is of course unavailable.

**Comment by Nivratti** [comment]
> > 在线演示和本地笔记本之间掩码差异的主要原因是，在线演示使用不同的模型来实现“分割所有内容”功能。在线演示由托管在以下终端节点的模型提供支持：
> > [segment_everything_box_model](https://model-zoo.metademolab.com/predictions/segment_everything_box_model)
> 
> The link you provided cannot be opened, why

The link I provided is an API reference URL that I identified by inspecting and analyzing the web application's code. The online demo model named `segment_everything_box_model` is hosted on private servers of meta and isn't publicly accessible. Online demos interact with this private model via API calls.
